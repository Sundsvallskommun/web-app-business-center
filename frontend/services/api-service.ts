import Router from 'next/router';
import { apiURL } from '@utils/api-url';
import axios from 'axios';
import { ApiResponseMeta } from '@interfaces/service';

export interface Data {
  error?: string;
  _meta?: ApiResponseMeta;
}

export interface ApiResponse<T> {
  data: T;
  message: string;
}

let isChangingRoute = false;
let isRedirectingToLogin = false;

Router.events.on('routeChangeStart', (route) => {
  isChangingRoute = true;
});

Router.events.on('routeChangeComplete', (route) => {
  isRedirectingToLogin = false;
  isChangingRoute = false;
  if (route === '/login') {
    isRedirectingToLogin = false;
  }
});

export const handleError = (error) => {
  /*let s = '';
  if (error.response) {
    // The request was made and the server responded with a status code
    // that falls out of the range of 2xx
    s += `Server responded with ${error.response.status} ${error.response.data.message}`;
  } else if (error.request) {
    // The request was made but no response was received
    // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
    // http.ClientRequest in node.js
    s += `Server did not respond`;
  } else {
    // Something happened in setting up the request that triggered an Error
    s += `Unknown error: ${error.message}`;
  }
  s += ` for url ${error.config.url}`;
  */
  // console.error(s);

  // FIXME: we should have a logged in guard?
  if (error.response.status === 401 && Router.pathname !== '/login' && !isRedirectingToLogin) {
    isRedirectingToLogin = true;
    Router.push('/login');
  }

  throw error;
};

const options = {
  headers: {
    'Content-Type': 'application/json',
  },
  withCredentials: true,
};

const get = <T>(url: string) => axios.get<T>(apiURL(url), options).catch(handleError);

const post = <T>(url: string, data: any) => {
  return axios.post<T>(apiURL(url), data, options).catch(handleError);
};

const remove = <T>(url: string) => {
  return axios.delete<T>(apiURL(url), options).catch(handleError);
};

const patch = <T>(url: string, data: any) => {
  return axios.patch<T>(apiURL(url), data, options).catch(handleError);
};

export const apiService = { get, post, patch, delete: remove };
