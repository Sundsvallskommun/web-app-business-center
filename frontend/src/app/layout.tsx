import 'dayjs/locale/se';
import '../../tailwind.scss';
import MyAppLayout from '../layouts/app/layout.component';

export default function Layout({ children }) {
  return <MyAppLayout>{children}</MyAppLayout>;
}
