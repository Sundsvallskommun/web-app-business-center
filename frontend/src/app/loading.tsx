'use client';

import FullscreenMainSpinner from '../components/spinner/fullscreen-main-spinner.component';

export default function Loading() {
  return <FullscreenMainSpinner />;
}
