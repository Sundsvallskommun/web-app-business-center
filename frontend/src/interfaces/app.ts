import { BusinessInformation } from './organisation-info';

export enum RepresentingMode {
  PRIVATE,
  BUSINESS,
}

export interface RepresentingEntityDto {
  organizationNumber?: string;
  personNumber?: string;
  mode?: RepresentingMode;
}
export interface RepresentingEntity {
  PRIVATE?: RepresentingPrivateEntity;
  BUSINESS?: RepresentingBusinessEntity;
  mode?: RepresentingMode;
}

export interface RepresentingBusinessEntity {
  organizationName: string;
  organizationNumber: string;
  information: BusinessInformation;
}

export interface RepresentingPrivateEntity {
  name: string;
}
