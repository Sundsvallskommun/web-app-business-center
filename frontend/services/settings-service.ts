import { ApiResponse, apiService, Data } from './api-service';

export enum FeedbackLifespan {
  'untilRemoved' = 'untilRemoved',
  'twoWeeks' = 'twoWeeks',
  'oneMonth' = 'oneMonth',
}

export interface ContactChannel {
  contactMethod: string;
  destination: string;
  sendFeedback: boolean;
  alias: string;
}

export const defaultChannel: ContactChannel = {
  contactMethod: '',
  destination: '',
  sendFeedback: true,
  alias: '',
};

export type ContactSettingsResponse = Array<{
  id: string;
  partyId: string;
  contactChannels: ContactChannel[];
  created: string;
  modified: string;
}>;

export interface ContactSettingsUpdateModel {
  id: string;
  contactChannels: ContactChannel[];
}

export interface OverviewFormModel {
  feedbackLifespan: FeedbackLifespan;
}

export const defaultOverviewsSettings: OverviewFormModel = {
  feedbackLifespan: FeedbackLifespan.oneMonth,
};

export interface Contact {
  alias: string;
  contactMethods: {
    SMS: ContactChannel[];
    EMAIL: ContactChannel[];
  };
}

export const defaultContact = {
  alias: '',
  contactMethods: {
    SMS: [],
    EMAIL: [],
  },
};

export interface ContactFormModel {
  id: string;
  contacts: Contact[];
}

export interface FeedbackData extends Data {
  settings: ContactFormModel;
}

export const defaultContactSettings: ContactFormModel = {
  id: '',
  contacts: [],
};

export const handleContactResponse: (res: ApiResponse<ContactSettingsResponse>) => ContactFormModel = (res) => ({
  id: res.data[0].id,
  contacts: Object.values(
    res.data.reduce((accumulator, contactSettingsGroup) => {
      contactSettingsGroup.contactChannels.forEach((channel) => {
        if (!accumulator[channel.alias]) {
          accumulator[channel.alias] = {
            alias: channel.alias,
            contactMethods: {
              SMS: [],
              EMAIL: [],
            },
          };
        }
        if (channel['contactMethod'] == 'SMS') {
          accumulator[channel.alias].contactMethods['SMS'].push(channel);
        }
        if (channel['contactMethod'] == 'EMAIL') {
          accumulator[channel.alias].contactMethods['EMAIL'].push(channel);
        }
      });
      return accumulator;
    }, {})
  ),
});

export const getContactSettings: () => Promise<FeedbackData> = () =>
  apiService
    .get<ApiResponse<ContactSettingsResponse>>('contactsettings')
    .then((res) => ({ settings: handleContactResponse(res.data) } as FeedbackData))
    .catch((e) => ({ settings: defaultContactSettings, error: e.response?.status ?? 'UNKNOWN ERROR' } as FeedbackData));

export const makeContactSettingsUpdateModel: (settings: ContactFormModel) => ContactSettingsUpdateModel = (
  settings
) => {
  return {
    id: settings.id,
    contactChannels: [
      ...settings.contacts
        .reduce((acc, curr) => {
          acc.push(curr.contactMethods['SMS']);
          acc.push(curr.contactMethods['EMAIL']);

          return acc;
        }, [])
        .flat(2),
    ],
  };
};

export const newContactSettings: (settings: ContactFormModel) => Promise<boolean> = (settings) => {
  return apiService
    .post('contactsettings', makeContactSettingsUpdateModel(settings))
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};

export const saveContactSettings: (settings: ContactFormModel) => Promise<boolean> = (settings) => {
  return apiService
    .patch('contactsettings', makeContactSettingsUpdateModel(settings))
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};
