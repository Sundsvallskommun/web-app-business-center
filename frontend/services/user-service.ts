import { FeedbackLifespan, User } from '@interfaces/user';
import dayjs from 'dayjs';
import { ApiResponse, apiService } from './api-service';

export const emptyUser: User = {
  name: '',
  userSettings: {
    feedbackLifespan: FeedbackLifespan.oneMonth,
    readNotificationsClearedDate: dayjs().toISOString(),
  },
};

const emptyUserResponse: ApiResponse<User> = {
  data: emptyUser,
  message: 'none',
};

export interface UserMetaResponse {
  lastLoginTime?: string;
}

const mockUserMeta: UserMetaResponse = {
  lastLoginTime: dayjs().subtract(180, 'day').toISOString(),
};

// TODO Return actual data when available in api
export const getUserMeta: () => Promise<UserMetaResponse> = () => Promise.resolve(mockUserMeta);

const handleSetUserResponse: (res: ApiResponse<User>) => User = (res) => ({
  name: res.data.name,
  userSettings: {
    feedbackLifespan: res.data.userSettings.feedbackLifespan,
    readNotificationsClearedDate: res.data.userSettings.readNotificationsClearedDate,
  },
});

export const getMe: () => Promise<User> = () => {
  return apiService
    .get<ApiResponse<User>>('me')
    .then((res) => handleSetUserResponse(res.data))
    .catch(() => Promise.reject('No user found'));
};

export const saveUserSettings: (settings: any) => Promise<boolean> = (settings) => {
  return apiService
    .patch('settings', settings)
    .then(() => Promise.resolve(true))
    .catch((e) => {
      return Promise.resolve(false);
    });
};
