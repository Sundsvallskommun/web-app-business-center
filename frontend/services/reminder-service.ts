import dayjs from 'dayjs';
import { statusCodes } from '../interfaces/status-codes';
import { ApiResponse, apiService, Data } from './api-service';

interface ReminderResponse {
  reminderId: string;
  personId: string;
  action: string;
  note: string;
  caseId: string;
  caseLink: string;
  reminderDate: string;
  caseType: string;
  externalCaseId: string;
  created: string;
  createdBy: string;
  modified: string;
  modifiedBy: string;
}

export type SendNoteOmittedTypes = 'reminderId' | 'personId' | 'created' | 'createdBy' | 'modified' | 'modifiedBy';
export type SendReminder = Omit<ReminderResponse, SendNoteOmittedTypes>;

export interface ReminderLabel {
  label: string;
  sortable: boolean;
  shownForStatus: statusCodes;
  screenReaderOnly: boolean;
}

export interface RemindersData extends Data {
  reminders: ReminderFormModel[];
  labels: ReminderLabel[];
}

export interface ReminderFormModel {
  heading: string;
  note: string;
  caseId: string;
  caseLink: string;
  reminderDate: string;
  caseType: string;
  externalCaseId: string;
  isReminder?: boolean;
  id?: string;
  created?: string;
  createdBy?: string;
  modified?: string;
  modifiedBy?: string;
}

export const defaultReminder = {
  heading: '',
  note: '',
  caseId: '',
  caseLink: '',
  reminderDate: '',
  caseType: '',
  externalCaseId: '',
  isReminder: false,
};

export interface ReminderEditFormModel extends ReminderFormModel {
  reminderId: string;
}

export const remindersLabels = [
  { label: 'Ärende', screenReaderOnly: false, sortable: true, shownForStatus: statusCodes.Any },
  { label: 'Senast ändrat', screenReaderOnly: false, sortable: true, shownForStatus: statusCodes.Any },
  { label: 'Skapad av', screenReaderOnly: false, sortable: true, shownForStatus: statusCodes.Any },
  { label: 'Anteckning', screenReaderOnly: false, sortable: true, shownForStatus: statusCodes.Any },
  { label: 'Ärendeknapp', screenReaderOnly: true, sortable: false, shownForStatus: statusCodes.Any },
];

export const emptyReminderList: RemindersData = { reminders: [], labels: remindersLabels };

export const handleReminderResponse: (res: ApiResponse<ReminderResponse[]>) => ReminderFormModel[] = (res) =>
  res.data.map((n: ReminderResponse, idx) => ({
    heading: n.action,
    note: n.note,
    caseId: n.caseId,
    caseLink: n.caseLink,
    reminderDate: n.reminderDate,
    caseType: n.caseType,
    externalCaseId: n.externalCaseId,
    isReminder: true,
    id: n.reminderId,
    created: dayjs(n.created).format('YYYY-MM-DD HH:MM'),
    createdBy: n.createdBy,
    modified: dayjs(n.modified).format('YYYY-MM-DD HH:MM'),
    modifiedBy: n.modifiedBy,
  }));

export interface Action {
  id: number;
  label: string;
  enabled: boolean;
}

export const actions: Action[] = [
  { id: 1, label: 'Följa upp', enabled: true },
  { id: 2, label: 'Komplettera', enabled: true },
  { id: 3, label: 'Förnya tillstånd', enabled: true },
  { id: 4, label: 'Skicka nytt ärende', enabled: true },
  { id: 5, label: 'Övrigt', enabled: true },
];

export const ongoingCasesActions: Action[] = [
  { id: 1, label: 'Följa upp', enabled: true },
  { id: 2, label: 'Komplettera', enabled: true },
  { id: 3, label: 'Övrigt', enabled: true },
];

export const closedCasesActions: Action[] = [
  { id: 1, label: 'Följa upp', enabled: true },
  { id: 2, label: 'Förnya tillstånd', enabled: true },
  { id: 3, label: 'Skicka nytt ärende', enabled: true },
  { id: 4, label: 'Övrigt', enabled: true },
];

export const getReminders: () => Promise<RemindersData> = () =>
  apiService
    .get<ApiResponse<ReminderResponse[]>>('reminders')
    .then((res) => ({ reminders: handleReminderResponse(res.data), labels: remindersLabels } as RemindersData))
    .catch((e) => ({ ...emptyReminderList, error: e.response?.status ?? 'UNKNOWN ERROR' } as RemindersData));

export const handleSendReminder: (reminder: ReminderFormModel) => SendReminder = (reminder) => ({
  action: reminder.heading,
  note: reminder.note,
  caseId: reminder.caseId,
  caseLink: reminder.caseLink,
  caseType: reminder.caseType,
  externalCaseId: reminder.externalCaseId ? reminder.externalCaseId.toString() : reminder.externalCaseId,
  reminderDate: reminder.reminderDate,
});

export const saveReminder: (reminder: ReminderFormModel) => Promise<boolean> = (reminder) => {
  return apiService
    .post('reminders', handleSendReminder(reminder))
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};

export const editReminder: (reminder: ReminderFormModel) => Promise<boolean> = (reminder) => {
  return apiService
    .patch(`reminders/${reminder.id}`, handleSendReminder(reminder))
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};

export const deleteReminder: (id: string) => Promise<boolean> = (id) => {
  return apiService
    .delete(`reminders/${id}`)
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};
