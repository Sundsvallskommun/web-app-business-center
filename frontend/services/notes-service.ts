import dayjs from 'dayjs';
import { statusCodes } from '../interfaces/status-codes';
import { ApiResponse, apiService, Data } from './api-service';
import { ReminderFormModel } from './reminder-service';
import { ApiResponseMeta } from '@interfaces/service';

export interface NoteResponse {
  id: string;
  subject: string;
  body: string;
  caseId: string;
  caseLink: string;
  caseType: string;
  externalCaseId: string;
  createdBy?: string;
  created: string;
  modified: string;
  modifiedBy: string;
  context: string;
  role: string;
  clientId: string;
}

const emptyNotesResponse: ApiResponse<{ notes: NoteResponse[] }> = {
  data: { notes: [] },
  message: 'none',
};

export type SendNoteOmittedTypes = 'id' | 'partyId' | 'created' | 'modified' | 'modifiedBy';

export interface SendNote {
  subject: string;
  body: string;
  caseId: string;
  caseLink: string;
  caseType: string;
  externalCaseId: string;
  createdBy?: string;
}

export interface NoteLabel {
  label: string;
  sortable: boolean;
  shownForStatus: statusCodes;
  screenReaderOnly: boolean;
}

export interface NotesData extends Data {
  _meta?: ApiResponseMeta;
  notes: NoteResponse[];
  labels: NoteLabel[];
}

export const noteLabels: NoteLabel[] = [
  { label: 'Ärende', sortable: false, screenReaderOnly: true, shownForStatus: statusCodes.Any },
  { label: 'Innehåll', sortable: false, screenReaderOnly: true, shownForStatus: statusCodes.Any },
];

export const emptyNotesList: NotesData = { notes: [], labels: noteLabels };

export const handleNotesResponse: (res: ApiResponse<{ notes: NoteResponse[] }>) => ReminderFormModel[] = (res) =>
  res.data.notes.map((n: NoteResponse) => ({
    heading: n.subject,
    note: n.body,
    caseId: n.caseId,
    caseLink: n.caseLink,
    reminderDate: '',
    caseType: n.caseType,
    externalCaseId: n.externalCaseId,
    isReminder: false,
    id: n.id,
    created: dayjs(n.created).format('YYYY-MM-DD HH:MM'),
    createdBy: n.createdBy,
    modified: dayjs(n.modified).format('YYYY-MM-DD HH:MM'),
    modifiedBy: n.modifiedBy,
  }));

export const getNotes: () => Promise<NotesData> = () =>
  apiService
    .get<ApiResponse<{ notes: NoteResponse[]; _meta: ApiResponseMeta }>>('notes')
    .then(
      (res) =>
        ({
          notes: handleNotesResponse(res.data),
          labels: noteLabels,
          _meta: res.data.data._meta,
        } as unknown as NotesData)
    )
    .catch(
      (e) =>
        ({
          notes: handleNotesResponse(emptyNotesResponse),
          labels: noteLabels,
          error: e.response?.status ?? 'UNKNOWN ERROR',
        } as unknown as NotesData)
    );

export const handleSendNote: (reminder: ReminderFormModel) => SendNote = (reminder) => ({
  subject: reminder.heading,
  body: reminder.note,
  caseId: reminder.caseId,
  caseLink: reminder.caseLink,
  caseType: reminder.caseType,
  externalCaseId: reminder.externalCaseId,
});

export const saveNote: (note: ReminderFormModel) => Promise<boolean> = (note) => {
  return apiService
    .post('notes', handleSendNote(note))
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};

export const editNote: (note: ReminderFormModel) => Promise<boolean> = (note) => {
  return apiService
    .patch(`notes/${note.id}`, handleSendNote(note))
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};

export const deleteNote: (id: string) => Promise<boolean> = (id) => {
  return apiService
    .delete(`notes/${id}`)
    .then(() => Promise.resolve(true))
    .catch((e) => Promise.resolve(false));
};
