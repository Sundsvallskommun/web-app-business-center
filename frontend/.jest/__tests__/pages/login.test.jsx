import '@jestRoot/__mocks__/services.mock';
import '@jestRoot/__mocks__/context.mock';
import { render, screen, act, waitFor } from '@testing-library/react';
import { AppWrapper } from '@contexts/app.context';
import * as nextRouter from 'next/router';
import Login from '@pages/login';

nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({ route: '/login', push: jest.fn() }));

describe('Login', () => {
  let container;
  beforeEach(async () => {
    container = await render(
      <AppWrapper>
        <Login />
      </AppWrapper>
    );
  });

  it('renders Login unchanged', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders Login and checks for h1', async () => {
    expect(container.queryByRole('heading', { level: 1 })).toBeTruthy();
  });

  it('renders Login and checks for main', async () => {
    expect(container.queryByRole('main')).toBeTruthy();
  });
});
