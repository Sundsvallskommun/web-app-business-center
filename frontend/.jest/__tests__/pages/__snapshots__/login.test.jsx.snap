// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders Login unchanged 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div
      class="Toaster"
      id="react-toast"
    />
    <div>
      <div
        class="bg-gray-lighter min-h-screen"
      >
        <main>
          <div
            class="flex items-center justify-center min-h-screen"
          >
            <div
              class="max-w-5xl w-full flex flex-col bg-white p-20 shadow-lg text-left"
            >
              <div
                class="mb-14"
              >
                <h1
                  class="mb-10 text-xl"
                >
                  Mina sidor företag
                </h1>
                <p
                  class="my-0"
                >
                  I Mina sidor för företagare finns en översikt av de ärenden som ditt företag har kopplat till Sundsvalls kommun. För att kunna se företagets ärenden behöver du vara firmatecknare, styrelseledamot eller revisor.
                </p>
              </div>
              <button
                class="btn btn-md btn-solid"
                data-color="primary"
                data-cy="loginButton"
              >
                Logga in
              </button>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-gray-lighter min-h-screen"
    >
      <main>
        <div
          class="flex items-center justify-center min-h-screen"
        >
          <div
            class="max-w-5xl w-full flex flex-col bg-white p-20 shadow-lg text-left"
          >
            <div
              class="mb-14"
            >
              <h1
                class="mb-10 text-xl"
              >
                Mina sidor företag
              </h1>
              <p
                class="my-0"
              >
                I Mina sidor för företagare finns en översikt av de ärenden som ditt företag har kopplat till Sundsvalls kommun. För att kunna se företagets ärenden behöver du vara firmatecknare, styrelseledamot eller revisor.
              </p>
            </div>
            <button
              class="btn btn-md btn-solid"
              data-color="primary"
              data-cy="loginButton"
            >
              Logga in
            </button>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
